pkg/plugins/localfiles.go

```diff
@@ -141,13 +142,14 @@ func (f LocalFS) Files() ([]string, error) {
 	// Convert the accumulator into a slice of relative path strings
 	relFiles := make([]string, 0, len(absFilePaths))
 	base := f.Base()
+	upperLevelPrefix := ".." + string(os.PathSeparator)
 	for fn := range absFilePaths {
 		relPath, err := filepath.Rel(base, fn)
 		if err != nil {
 			return nil, err
 		}
 		clenRelPath, err := util.CleanRelativePath(relPath)
-		if strings.Contains(clenRelPath, "..") || err != nil {
+		if strings.HasPrefix(clenRelPath, upperLevelPrefix) || err != nil {
```

Comment:
`Contains` -> `HasPrefix` is be safe here.

`CleanRelativePath` cleans the whole path, so `../` can only be present at the beginning and not anywhere else in the string:

https://go.dev/play/p/WsRuQ407YPB.go

Reply:
is this `HasPrefix` even needed? As far as I can see, `clean` is already removing it:

https://play.golang.com/p/qAcoWjnIOnE

Reply:
Yeah `CleanRelativePath` is adding `/` at the beginning, so it will never be above root and the HasPrefix is always false. It can be removed.

However, `CleanRelativePath` is always using `/` as path separator, so I am not sure if it (in general) works properly on Windows...? :thinking: @wbrowne any ideas?

```go
// CleanRelativePath returns the shortest path name equivalent to path
// by purely lexical processing. It make sure the provided path is rooted
// and then uses filepath.Clean and filepath.Rel to make sure the path
// doesn't include any separators or elements that shouldn't be there
// like ., .., //.
func CleanRelativePath(path string) (string, error) {
	cleanPath := filepath.Clean(filepath.Join("/", path))
	rel, err := filepath.Rel("/", cleanPath)
	if err != nil {
		// slash is prepended above therefore this is not expected to fail
		return "", err
	}

	return rel, nil
}
```

Reply:
I suspect that should use `path.Separator` (but nothing for this PR)

Reply:
I believe so. I will do some further testing on Windows later and open a follow-up PR.

Reply:
Yeah I thought we looked at this recently regarding the renderer issue but will take a look again